flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/DynamicTableFactory.java

```diff
@@ -85,5 +86,24 @@
 
         /** Whether the table is temporary. */
         boolean isTemporary();
+
+        /**
+         * Shortcut for {@code getCatalogTable().getResolvedSchema().toPhysicalRowDataType()}.
```

Comment:
I think we should add instructional information here on what to use this for / how to use it in the specific context of the connector (since we offer a shortcut for it). This method will be called by basically every connector implementation.